Code/DataStructs/Utils.cpp

```diff
@@ -330,16 +330,21 @@ void a2b(const char *a4, char *b3) {
     *** b3: |000000111111222222333333
     ***     |=====+=====+=====+=====|
     *********************************************/
-    if (i == 0) {
-      b3[0] = (byte << 2); /*** 6 bits into 1st byte ***/
-    } else if (i == 1) {
-      b3[0] |= ((b = byte) >> 4); /*** 2 bits into 1st byte ***/
-      b3[1] = ((b = byte) << 4);  /*** 4 bits into 2nd byte ***/
-    } else if (i == 2) {
-      b3[1] |= ((b = byte) >> 2); /*** 4 bits into 2nd byte ***/
-      b3[2] = ((b = byte) << 6);  /*** 2 bits into 3rd byte ***/
-    } else if (i == 3) {
-      b3[2] |= byte; /*** 6 bits into 3rd byte ***/
+    switch (i) {
+      case 0:
+        b3[0] = (byte << 2); /*** 6 bits into 1st byte ***/
+        break;
+      case 1:
+        b3[0] |= ((b = byte) >> 4); /*** 2 bits into 1st byte ***/
+        b3[1] = ((b = byte) << 4);  /*** 4 bits into 2nd byte ***/
+        break;
+      case 2:
+        b3[1] |= ((b = byte) >> 2); /*** 4 bits into 2nd byte ***/
+        b3[2] = ((b = byte) << 6);  /*** 2 bits into 3rd byte ***/
+        break;
+      case 3:
```

Comment:
You need a `default` in the switch statement

Reply:
In the function `i` can only ever be 0 to 3. In principle it should never reach `default`.
Is it best-practice for this code-base to always have a `default` in switch statements? In this case I would add one also to the immediately preceding switch statement.

Reply:
We don't actually have a "standard" for this. If one of the compilers warns about the "missing" default, then the build will fail and you'll need to add it, but as long as the builds are passing then it's not a problem. I guess in this case the compilers didn't complain about your original construction, so it was fine.